Add instructions on how to install plasmapy with uv in a virtual envi… (#2861)
PlasmaPy/PlasmaPy: docs/install.rst

For more detailed information, please refer to this tutorial on
`installing packages`_.

.. _install-uv:

Installing PlasmaPy with uv
===========================

|uv| is "an extremely fast |Python| package and project manager, written
in Rust". |uv| lets us create and switch between |Python| environments
that are isolated from each other and the system's |Python|
installation. In addition, |uv| provides a `pip drop-in interface`_ for
common |pip| commands so that |Python| packages on |PyPI| may be
installed into |uv|-managed
`virtual environments <virtual environment_>`_ without installing |pip|.

In a virtual environment
------------------------

After `installing uv`_, a `virtual environment`_ with |Python| version
|maxpython| can be created by opening a terminal and running:

.. code-block:: bash

   uv venv --python |maxpython|

> |uv| will automatically download the |Python| binary and symlink it to

Reviewer: ```suggestion
|uv| will automatically download |Python| and link it to
```
🤔 I'm wondering if this would be a little more accessible to people who are still new to Python.

> the `virtual environment`_'s directory at (by default) ``.venv``. The

Reviewer: ```suggestion
the `virtual environment`_'s directory at (by default) :file:`.venv`. The
```
`:file:` is a reStructuredText role that allows us to have a slightly different text style when referring to files, though it doesn't link to the actual file.

environment can then be activated by running:

.. code-block:: bash

   source .venv/bin/activate

Then, to install `plasmapy` into the activated environment, run:

.. code-block:: bash

   uv pip install plasmapy

> In a python project

Reviewer: ```suggestion
In a Python project
```
🎨 also for consistency

-------------------

|uv| also supports managing |Python| projects. A `Python project`_
is a collection of |Python| code or package(s) and additional files,

> e.g. ``pyproject.toml``, which contains the `project metadata`_. To make
> a `Python project`_ reproducible across many platforms, |uv| reads
> ``pyproject.toml`` and generates a lockfile name ``uv.lock``, which in

Reviewer: ```suggestion
e.g. :file:`pyproject.toml`, which contains the `project metadata`_. To make
a `Python project`_ reproducible across many platforms, |uv| reads
:file:`pyproject.toml` and generates a lockfile name :file:`uv.lock`, which in
```

turn can be read by |uv| on a different machine to create the same
`Python project`_ and its dependencies.

To initialize a `Python project`_ named ``project-with-plasmapy`` with
|Python| version |maxpython|, run with |uv|:

.. code-block:: bash

>    uv init --name project-with-plasmapy --python |maxpython|

Reviewer: This capability would have made our lives a lot easier when starting PlasmaPy in the first place! 😅

> By default, |uv| will generate a ``pyproject.toml`` and many other

Reviewer: ```suggestion
By default, |uv| will generate a :file:`pyproject.toml` and many other
```

metadata files (see `uv init options`_). Then, to add `plasmapy` as a
dependency, run:

.. code-block:: bash

   uv add plasmapy

> |uv| will add `plasmapy` to ``pyproject.toml`` and create a `virtual
> environment`_ in ``.venv`` with `plasmapy` and its dependencies already

Reviewer: ```suggestion
|uv| will add `plasmapy` to :file:`pyproject.toml` and create a `virtual
environment`_ in :file:`.venv` with `plasmapy` and its dependencies already
```

> installed. So, it is no longer needed to use the `pip drop-in

Reviewer: 🤔 I didn't know that `uv add` and `uv remove` update the virtual environment too!  I thought it only updated `pyproject.toml` and `uv.lock`.  This will be quite useful.

interface`_. To activate the |Python| environment with `plasmapy`
installed, run:

.. code-block::bash

   source .venv/bin/activate

> 

Reviewer: 🌠 This would be out-of-scope for this PR and more of a wish list item, but would it make sense to eventually include this section in a new page in the [contributor guide](https://docs.plasmapy.org/en/latest/contributing) that discusses how to create an affiliated package? 🤔

Author: Good enough for me. It's your call. It may also be revisited in #2852.
